Sources/MongoSwift/MongoCollection+Indexes.swift

```diff
@@ -306,13 +327,20 @@ extension MongoCollection {
      *
      * - Returns: A `MongoCursor` over the index names.
      *
-     * - Throws: A `userError.invalidArgumentError` if the options passed are an invalid combination.
+     * - Throws: `UserError.logicError` if the provided session is inactive.
      */
-    public func listIndexes() throws -> MongoCursor<Document> {
-        guard let cursor = mongoc_collection_find_indexes_with_opts(self._collection, nil) else {
+    public func listIndexes(session: ClientSession? = nil) throws -> MongoCursor<Document> {
+        var opts: Document?
+        if let session = session {
```

Comment:
Is this the only place we have a method without options? Perhaps we should encapsulate this in some sort of utility method

Reply:
Updated this to use the new combine helper. It has a weird cast that needs to be there so the compiler can infer the generic type, even though it will be nil.